DPA-SJSU/Degree-Planning-Assistant
# [finish impl] [Feature] Add Email Verification

**Purpose of Tasks:**
- [Feature] [Register] Add Mailing feature. [file here](https://github.com/DPA-SJSU/Degree-Planning-Assistant/pull/77/files#diff-f8ce403c5d71e091ddb8dbf40fa39c8b)
- [Template] Adding [ejs](https://ejs.co/) templates for sending email or used for backend UI
- [Feature] Add [Subscription](https://github.com/DPA-SJSU/Degree-Planning-Assistant/pull/77/files#diff-b3429d26cbac3290829435c6d1b3e946)
- [Login] Add email verified Handler when user login (not log in when the email is not verified)
- Cleanup code

Step to test:
1. Call DELETE: http://localhost:8080/
![image](https://user-images.githubusercontent.com/19829770/82585895-9ac9f280-9b4b-11ea-8bac-1f5f61cea7f4.png)

2. Register with your real email
![image](https://user-images.githubusercontent.com/19829770/82585923-a4ebf100-9b4b-11ea-9941-c8998992d55f.png)

3. Test Login without verifying email
![image](https://user-images.githubusercontent.com/19829770/82585972-baf9b180-9b4b-11ea-9431-b3b19bcd775d.png)

4. Open email
![image](https://user-images.githubusercontent.com/19829770/82586019-c9e06400-9b4b-11ea-94f1-7f115f716550.png)

5. Click on the verify link 
![image](https://user-images.githubusercontent.com/19829770/82586182-03b16a80-9b4c-11ea-9091-60b88c4015f5.png)

6. Automatic redirect to client front-end
